File: examples/duna-service-example/src/impl/java/io/duna/example/EchoServiceImpl.java
/*
 * Copyright (c) 2017 Duna Open Source Project
 * Ministério do Planejamento, Desenvolvimento de Gestão
 * República Federativa do Brasil
 *
 * This file is part of the Duna Project.
 */
package io.duna.example;

import io.duna.core.service.Service;
import io.duna.http.Parameter;

@Service
public class EchoServiceImpl implements EchoService {

    /*
     * The @Parameter annotation is optional when compiled with Java 8's argument -parameter
     */
    @Override
    public String echo(@Parameter("shout") String shout) {
        return shout;
    }
}

File: examples/src/echo/kotlin/io/duna/example/echo/impl/EchoServiceImpl.kt
/*
 * Copyright (c) 2016 Duna Open Source Project
 * Ministério do Planejamento, Desenvolvimento de Gestão
 * República Federativa do Brasil
 *
 * This file is part of the Duna Project.
 */
package io.duna.example.echo.impl

import io.duna.core.service.Service
import io.duna.example.echo.EchoService
import io.duna.http.HttpMethod
import io.duna.http.HttpPort
import io.duna.http.Verb

@Service
@HttpPort
open class EchoServiceImpl : EchoService {

  @HttpMethod(Verb.GET)
  override fun echo(shout: String, myAsf: Int): String {
    return shout
  }
}
